C-Sharp/graphs/DFS.cs

```diff
@@ -0,0 +1,182 @@
+        static void GraphTraversal(Dictionary<string, Vertex> graph)
+        {
+            Stack<Vertex> traverseStack = new Stack<Vertex>();
+            Vertex unvisited;
+            while ((unvisited = FindUntraversedVertex(graph)) != null)
+            {
+                traverseStack.Push(unvisited);
+                while (traverseStack.Count > 0)
+                {
+                    Vertex vertex = traverseStack.Pop();
+                    vertex.Traversed = true;
+                    Console.Write($"{vertex.Value} ");
+
+                    foreach (Edge incidentEdge in vertex.Edges)
+                    {
+                        if (!incidentEdge.EndVertex.Traversed)
+                        {
+                            traverseStack.Push(incidentEdge.EndVertex);
+                        }
+                    }
+                }
+                Console.WriteLine();
+            }
+        }
+
+        /// <summary>
+        /// In case of disconnected graph, for the generation of DFS forest(instead of single tree) this method find untraversed vertices 
+        /// </summary>
+        /// <param name="graph">Dictionary representing the graph</param>
+        /// <returns></returns>
+        static Vertex FindUntraversedVertex(Dictionary<string, Vertex> graph)
+        {
+            foreach (KeyValuePair<string, Vertex> graphVertice in graph)
+            {
+                if (!graphVertice.Value.Traversed)
+                {
+                    return graphVertice.Value;
+                }
+            }
+            return null;
+        }
+
+        /// <summary>
+        /// Resetting states of all vertices to unvisited after DFS traversal
+        /// </summary>
+        /// <param name="graph"></param>
+        static void ResetGraph(Dictionary<string, Vertex> graph)
+        {
+            foreach (KeyValuePair<string, Vertex> graphVertex in graph)
+            {
+                graphVertex.Value.Traversed = false;
+            }
+        }
+
+
+        /// <summary>
+        /// Vertex reprsents each vertex within the graph having three properteies 
+        /// Value represents actual name of the vertex
+        /// Traversed determines whether the vertex has already been traversed or not
+        /// Edges determines list of edges to denote edges adjacent to vertex. 
+        /// </summary>
+        private class Vertex
+        {
+            internal string Value { get; set; }
+            internal bool Traversed { get; set; }
+            internal List<Edge> Edges { get; set; }
+
+            internal Vertex(string value)
+            {
+                Value = value;
+                Edges = new List<Edge>();
+                Traversed = false;
+            }
+        }
+
+        /// <summary>
+        /// Edge encapsulates end vertex of an edge
+        /// </summary>
+        private class Edge
+        {
+            internal Vertex EndVertex { get; set; }
+        }
+    }
+}
```

Comment:
Add sample I/O and complexity

Reply:
Sample I/O and complexity are written as comment from line 7 to line 24

Reply:
Kindly shift it to the bottom

Reply:
This is us just trying to maintain standards.